code/datums/components/status_effect_listener.dm

```diff
@@ -0,0 +1,29 @@
+/datum/component/status_effect_listener
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+	var/list/effects = list()
+
+/datum/component/status_effect_listener/Initialize()
+	RegisterSignal(COMSIG_ATOM_EX_ACT, .proc/explosion)
+	RegisterSignal(COMSIG_MOVABLE_MOVED, .proc/movement)
+	RegisterSignal(COMSIG_LIVING_RESIST, .proc/resist)
+	RegisterSignal(COMSIG_LIVING_IGNITED, .proc/ignited)
+	RegisterSignal(COMSIG_LIVING_EXTINGUISHED, .proc/extinguished)
+
+/datum/component/status_effect_listener/proc/signal(var/sigtype)
+	for(var/datum/status_effect/effect in effects)
+		effect.receiveSignal(sigtype)
+
+/datum/component/status_effect_listener/proc/explosion()
+	signal("explosion")
+
+/datum/component/status_effect_listener/proc/movement()
+	signal("movement")
+
+/datum/component/status_effect_listener/proc/resist()
+	signal("resist")
+
+/datum/component/status_effect_listener/proc/ignited()
+	signal("ignited")
+
+/datum/component/status_effect_listener/proc/extinguished()
+	signal("extinguished")
```

Comment:
It receives a signal that calls a proc that calls a proc that loops through everything calling a proc lol

Reply:
I never claimed I was a smart woman.

Reply:
you'd need another proc to call that lol

Reply:
I have an idea but prob need to flesh it out (not at work).

I don't see this being used anywhere in the pr though? Am I missing it?

Reply:
Not used in this PR, this is just setting up the framework.

Reply:
> framework with 0 implementation

stop this is the code sheriff ur goin in the slammer dirty dan

Reply:
`there's some implementation you'll never catch me alive coppers`

Reply:
@ExcessiveUseOfCobblestone Just added implementation for it! Removed some snowflake code :>